### ISSUE TITLE
lightgbm.cv strange results in Random Forest mode

### ISSUE THREAD

**Issue by candalfigomoro** [microsoft/LightGBM#4206]
This is basically a continuation of this closed issue: https://github.com/microsoft/LightGBM/issues/3387

With lightgbm's cv method, if I use 300 trees in Random Forest mode, I get 300 scores.

In comparison, when I use xgboost's cv method in Random Forest mode, I get just 1 score (this naturally arises from the fact that in xgboost we have decoupled parameters for boosted/parallel trees, see https://github.com/microsoft/LightGBM/issues/3128).

What is the intepretation of the 300 scores? Is the 1st score the one I'd get with 1 tree and the 300th score the one I get with a Random Forest of 300 trees?

While for boosting iterations this kind of output makes sense, I think the interpretation of lightgbm's cv results in Random Forest mode is a bit confusing.

**Comment by shiyu1994** [comment]
@candalfigomoro Thanks for using LightGBM.

> What is the intepretation of the 300 scores? Is the 1st score the one I'd get with 1 tree and the 300th score the one I get with a Random Forest of 300 trees?

Yes, that's correct. The `i`th value in the list represents the averaged `mean` or `std` across the folds using the first `i` iterations.
This is because `rf` is treated as a special type of `boosting` in LightGBM, which does not update the gradients between iterations.

**Comment by candalfigomoro** [comment]
@shiyu1994 Thank you!
